### PR TITLE
Task/saml2 conf for idp

### PR DESCRIPTION
**Pull request by AlvaroVega**
Doc:
https://github.com/telefonicaid/fiware-keystone-spassword/blob/task/saml2_conf_for_idp/docs/iotp_saml_idp.md

New docker tag version: 1.12.0_saml2
https://hub.docker.com/layers/telefonicaiot/fiware-keystone-spassword/1.12.0_saml2/images/sha256-226d8c4349c62d14c2cf821a1a22f9a1b6b519f8e7a2be6d81f38fbbfa56a46b?context=repo

This is how use new env vars for docker-compose:
```
    - SAML_ENDPOINT=localhost:5001
    - SAML_CERTFILE=/etc/keystone/ssl/server.crt
    - SAML_KEYFILE=/etc/keystone/ssl/server.key
  volumes:
    - /home/iotp/server.crt:/etc/keystone/ssl/server.crt
    - /home/iotp/server.key:/etc/keystone/ssl/server.key
````

### CODE REVIEW

**Comment by AlvaroVega** [comment]
btw, this is how I create that certificate and key file for testing:
```
openssl genrsa -des3 -passout pass:xxxx -out server.pass.key 2048 
openssl rsa -passin pass:xxxx -in server.pass.key -out server.key
rm server.pass.key 
openssl req -new -key server.key -out server.csr \
    -subj "/C=ES/ST=DistritoT/L=Madrid/O=Telefonica/OU=Telefonica On premise platform/CN=telefonica.com"
openssl x509 -req -days 3000 -in server.csr -signkey server.key -out server.crt
```

File: docs/iotp_saml_idp.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+export OS_USER_DOMAIN_ID=default
+export OS_USER_DOMAIN_NAME=Default
+export OS_PROJECT_DOMAIN_ID=default
+export OS_PROJECT_DOMAIN_NAME=Default
+export OS_AUTH_URL=http://127.0.0.1:5001/v3
+export OS_IDENTITY_API_VERSION=3
+export OS_PASSWORD='<password>'
+```
+
+and then execute the followign command with proper urls:
```
Comment (fgalan) [inline docs/iotp_saml_idp.md:52]:
```suggestion
and then execute the following command with proper urls:
```

File: docs/iotp_saml_idp.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+- SAML_CERTFILE
+This env var will be expanden in /etc/keystone keystone.conf
+
+```
+[saml]
+certfile=/etc/keystone/ssl/certs/signing_cert.pem
+```
+
+- SAML_KEYFILE
+This env var will be expanden in /etc/keystone keystone.conf
```
Comment (fgalan) [inline docs/iotp_saml_idp.md:29]:
```suggestion
This env var will be expanded in /etc/keystone keystone.conf
```

File: docs/iotp_saml_idp.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+[saml]
+idp_entity_id=https://$SAML_ENDPOINT/v3/OS-FEDERATION/saml2/idp
+idp_sso_endpoint=https://$SAML_ENDPOINT/v3/OS-FEDERATION/saml2/sso
+```
+
+- SAML_CERTFILE
+This env var will be expanden in /etc/keystone keystone.conf
```
Comment (fgalan) [inline docs/iotp_saml_idp.md:21]:
```suggestion
This env var will be expanded in /etc/keystone/keystone.conf
```

(Not sure about the slash)

File: docs/iotp_saml_idp.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+One of the configuration models supported by Keystone for federated identity is  Keystone as Identity Provider (IdP).
+
+
+## Configuration
+Based on [Keystone as an Identity Provider](https://docs.openstack.org/keystone/stein/admin/federation/configure_federation.html#keystone-as-idp)
+
+These new DOCKER options should be used:
+
+- SAML_ENDPOINT
+This env var will be expanden in /etc/keystone keystone.conf
```
Comment (fgalan) [inline docs/iotp_saml_idp.md:12]:
```suggestion
This env var will be expanded in /etc/keystone/keystone.conf
```

(Not sure about the slash)

File: docs/iotp_saml_idp.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+- SAML_KEYFILE
+This env var will be expanden in /etc/keystone keystone.conf
+
+```
+[saml]
+keyfile=/etc/keystone/ssl/private/signing_key.pem
+```
+
+And then a Service Provider must be created and registered into IdP Keystone.
+To achive that get into container and set following env vars:
```
Comment (fgalan) [inline docs/iotp_saml_idp.md:37]:
```suggestion
To achive that, get into container and set following env vars:
```

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**